layerfetcher/layers.go

```diff
@@ -357,35 +363,33 @@ func (l *Logic) receiveLayerContent(ctx context.Context, layerID types.LayerID,
 // all blocks are fetched for a given layer.
 // it deliberately doesn't hold any lock while notifying channels.
 func notifyLayerBlocksResult(layerID types.LayerID, channels []chan LayerPromiseResult, lyrResult *layerResult, logger log.Log) {
-	var result *LayerPromiseResult
-	hasZeroBlockHash := false
-	var firstErr error
+	var (
+		missing, success bool
+		err              error
+	)
 	for _, res := range lyrResult.responses {
 		if res.err == nil && res.data != nil {
-			// at least one layer hash contains blocks. not a zero block layer
-			result = &LayerPromiseResult{Layer: layerID, Err: nil}
+			success = true
```

Comment:
is this necessary? i think you just need to check whether missing or err != nil outside the loop?

Reply:
if i only check the error then even a single failure will mark whole attempt failed